### PR TITLE
Update alert component for wallet connection

### PR DESCRIPTION
**Pull request by kkosiorowska**
Closes AENG-25
Closes https://github.com/thesis/acre/issues/571
Closes https://github.com/thesis/acre/issues/872

This PR updates the logic for displaying alerts for a wallet connection. What has been changed:

- Allowed to pass `status` to `ConnectWalletAlert`.
- Changed `description` type to `ReactNode` to create more custom content.
- Instead of passing title and description to the component, we just pass the type. This keeps the alert logic in one place, but also makes it easier to simply pass just one prop.
- Updated content for connection alerts.

### CODE REVIEW

File: dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        If the problem persists, contact{" "}
+        <Link
+          // TODO: Define in the new color palette
+          color="#0E61FE"
+          textDecoration="underline"
+          href={EXTERNAL_HREF.DISCORD}
+          isExternal
+        >
+          Acre support
+        </Link>
```
Comment (nkuba) [inline dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx:49]: Let's add a dot at the end of the sentence.
```suggestion
        </Link>.
```

File: dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+const CONNECTION_ALERTS = {
+  [ConnectionAlert.Rejected]: {
+    title: "Wallet connection rejected.",
+    description: "If you encountered an error, please try again.",
+  },
+  [ConnectionAlert.NotSupported]: {
+    title: "Not supported.",
+    description:
+      "Only Native SegWit, Nested SegWit or Legacy addresses supported at this time. Please try a different address or another wallet.",
```
[the remark this answers is not in the document]
Comment (kkosiorowska) [reply dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx]: https://github.com/thesis/acre/pull/901

File: dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          href={EXTERNAL_HREF.DISCORD}
+          isExternal
+        >
+          Acre support
+        </Link>
+      </Box>
+    ),
+  },
+  [ConnectionAlert.InvalidSIWWSignature]: {
+    title: "Invalid Sign In With Wallet signature",
```
[the remark this answers is not in the document]
Comment (kkosiorowska) [reply dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx]: https://github.com/thesis/acre/pull/901

File: dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+  },
+  [ConnectionAlert.NotSupported]: {
+    title: "Not supported.",
+    description:
+      "Only Native SegWit, Nested SegWit or Legacy addresses supported at this time. Please try a different address or another wallet.",
+  },
+  [ConnectionAlert.NetworkMismatch]: {
+    title: "Error!",
+    description:
+      "Incorrect network detected in your wallet. Please choose proper network and try again.",
```
[the remark this answers is not in the document]
Comment (kkosiorowska) [reply dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx]: https://github.com/thesis/acre/pull/901

File: dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+  },
+  [ConnectionAlert.Default]: {
+    title: "Wallet connection error. Please try again.",
```
[the remark this answers is not in the document]
Comment (kkosiorowska) [reply dapp/src/components/ConnectWalletModal/ConnectWalletAlert.tsx]: https://github.com/thesis/acre/pull/901

**Review by kpyszkowski — COMMENTED** [review]
LGTM. Leaving approval for @r-czajkowski

**Review by r-czajkowski — APPROVED** [review]

**Pull request merged**